### ISSUE TITLE
Cannot update to 5.1.3

### ISSUE THREAD

**Issue by SmashingQuasar** [pimcore/pimcore#2559]
Hello,

I can't update to Pimcore 5.1.3 from 5.1.2, I get the following exception:

```
 (1/1) ClassNotFoundException

Attempted to load class "PrestaSitemapBundle" from namespace "Presta\SitemapBundle".
Did you forget a "use" statement for another namespace?
```
The source code of Pimcore hasn't been edited at all, I just created objects in the back-office so far.

It seems now the installation of Pimcore is totally broken and I cannot load the back-office or any page.

Thanks in advance!

**Comment by dpfaffenbauer** [comment]
try running ```composer install```

**Comment by Cruiser13** [comment]
Same issue here, updating from 5.1.0 to 5.1.3.

**Comment by SmashingQuasar** [comment]
I'll just wait for a fix since I had a back-up of my data.

**Comment by dpfaffenbauer** [comment]
Have you guys tried what I said?

**Comment by SmashingQuasar** [comment]
I didn't, I simply restored my back-up, I need to continue working on this project for our client so I don't have enough time to try this sorry :(

**Comment by dpfaffenbauer** [comment]
There won't be a fix, you are missing dependencies that are installed thru composer....

**Comment by Cruiser13** [comment]
Hi @dpfaffenbauer I didn't investigate much so far but updating the same installation from 5.1.0 to 5.1.2 works fine. Composer is installed and visible in the requirements check. Of course installing dependencies again could work but I expect updates to go smoothly if the requirements are met. Update hangs up in the middle which should definately not be the case so I think this ticket is valid.

I'll post once I tried composer install again, working with the 5.1.2 at the moment.

**Comment by SmashingQuasar** [comment]
Considering I had a clean installation of Pimcore 5 and did every updates since its release without any issue, nor changing the source code, nor adding any custom bundle (basically I'm working on classes and import script from JSON into the PIM), I shouldn't miss any dependency. :(

**Comment by brusch** [comment]
I've just tried to reproduce the problem on various environments, but I wasn't able to reproduce the issue. 
What I've done: 
- installed build 190 (which some builds before 5.1.2) with demo data
- opened the web-updater in the admin interface
- started the update
- done 

Maybe someone can further investigate the problem and give us some hints how we can reproduce the problem.

**Comment by brusch** [comment]
@SmashingQuasar @Cruiser13 can you please apply the following patch and try the update again?
 https://github.com/pimcore/pimcore/commit/40585f9bfdb30f65213bca8ebc3e9d5c157cedbd

Additional questions: 
- What happens if you run `composer update` prior to the update? 
- In the case the update fails again, is `presta/sitemap-bundle` in your `composer.json` ? 
- If yes, is the code of the bundle at it's desired place? (`vendor/presta/sitemap-bundle`)?

**Comment by Cruiser13** [comment]
Hello @brusch ,
I just tried the update again. 5.1.0 to 5.1.2 went smoothly. Now I tried running 5.1.2 to 5.1.3. It failed after downloading at ~28% install process. I did not run composer update before the update though (did not read your message at this point).

My composer.json has this entry:
        "presta/sitemap-bundle": "^1.5"

The code of the bundle is at vendor/presta/sitemap-bundle.

This is my usagelog:

> 2018-02-07T14:26:47+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Admin\InstallController::checkAction|pimcore_admin_install_check|[]|[]
> 2018-02-07T14:34:58+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::checkFilePermissionsAction|pimcore_admin_update_index_checkfilepermissions|[]|[]
> 2018-02-07T14:34:59+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::checkDebugModeAction|pimcore_admin_update_index_checkdebugmode|[]|[]
> 2018-02-07T14:35:00+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::checkComposerInstalledAction|pimcore_admin_update_index_checkcomposerinstalled|[]|[]
> 2018-02-07T14:35:00+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::getAvailableUpdatesAction|pimcore_admin_update_index_getavailableupdates|[]|[]
> 2018-02-07T14:35:05+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Admin\InstallController::checkAction|pimcore_admin_install_check|[]|[]
> 2018-02-07T14:35:26+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::checkFilePermissionsAction|pimcore_admin_update_index_checkfilepermissions|[]|[]
> 2018-02-07T14:35:27+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::checkDebugModeAction|pimcore_admin_update_index_checkdebugmode|[]|[]
> 2018-02-07T14:35:28+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::checkComposerInstalledAction|pimcore_admin_update_index_checkcomposerinstalled|[]|[]
> 2018-02-07T14:35:28+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::getAvailableUpdatesAction|pimcore_admin_update_index_getavailableupdates|[]|[]
> 2018-02-07T14:35:32+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::getJobsAction|pimcore_admin_update_index_getjobs|[]|{"toRevision":"197"}
> 2018-02-07T14:35:33+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::jobParallelAction|pimcore_admin_update_index_jobparallel|[]|{"type":"download","revision":"193","url":"https:\/\/liveupdate.pimcore.org\/download-..."}
> 2018-02-07T14:35:33+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::jobParallelAction|pimcore_admin_update_index_jobparallel|[]|{"type":"download","revision":"195","url":"https:\/\/liveupdate.pimcore.org\/download-..."}
> 2018-02-07T14:35:34+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::jobParallelAction|pimcore_admin_update_index_jobparallel|[]|{"type":"download","revision":"194","url":"https:\/\/liveupdate.pimcore.org\/download-..."}
> 2018-02-07T14:35:34+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::jobParallelAction|pimcore_admin_update_index_jobparallel|[]|{"type":"download","revision":"195","url":"https:\/\/liveupdate.pimcore.org\/download-..."}
> 2018-02-07T14:35:34+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::jobParallelAction|pimcore_admin_update_index_jobparallel|[]|{"type":"download","revision":"196","url":"https:\/\/liveupdate.pimcore.org\/download-..."}
> 2018-02-07T14:35:34+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::jobParallelAction|pimcore_admin_update_index_jobparallel|[]|{"type":"download","revision":"197","url":"https:\/\/liveupdate.pimcore.org\/download-..."}
> 2018-02-07T14:35:39+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Admin\MiscController::maintenanceAction|pimcore_admin_misc_maintenance|[]|{"activate":"true"}
> 2018-02-07T14:35:40+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::jobProceduralAction|pimcore_admin_update_index_jobprocedural|[]|{"type":"composer-invalidate-classmaps"}
> 2018-02-07T14:35:40+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::jobProceduralAction|pimcore_admin_update_index_jobprocedural|[]|{"type":"files","revision":"193","updateScript":"false"}
> 2018-02-07T14:35:41+0100 : 2|Pimcore\Bundle\AdminBundle\Controller\Update\IndexController::jobProceduralAction|pimcore_admin_update_index_jobprocedural|[]|{"type":"files","revision":"194","updateScript":"false"}

This is my dev.log:


> [2018-02-07 14:35:41] app.DEBUG: Resolved pimcore context for path /admin/update/index/job-procedural to admin {"path":"/admin/update/index/job-procedural","context":"admin"} []
> [2018-02-07 14:35:41] security.DEBUG: Checking for guard authentication credentials. {"firewall_key":"admin","authenticators":1} []
> [2018-02-07 14:35:41] security.DEBUG: Calling getCredentials() on guard configurator. {"firewall_key":"admin","authenticator":"Pimcore\\Bundle\\AdminBundle\\Security\\Guard\\AdminAuthenticator"} []
> [2018-02-07 14:35:41] pimcore_admin.session.DEBUG: Opening admin session PHPSESSID {"name":"PHPSESSID"} []
> [2018-02-07 14:35:41] pimcore_admin.session.DEBUG: Admin session PHPSESSID was successfully opened. Open admin sessions: 1 {"name":"PHPSESSID","count":1} []
> [2018-02-07 14:35:41] pimcore_admin.session.DEBUG: Admin session PHPSESSID was written and closed {"name":"PHPSESSID"} []
> [2018-02-07 14:35:41] doctrine.DEBUG: SELECT * FROM users WHERE `type` = ? AND id = ? ["user","2"] []
> [2018-02-07 14:35:41] doctrine.DEBUG: SELECT * FROM users_workspaces_asset WHERE userId = ? [2] []
> [2018-02-07 14:35:41] doctrine.DEBUG: SELECT * FROM users_workspaces_document WHERE userId = ? [2] []
> [2018-02-07 14:35:41] doctrine.DEBUG: SELECT * FROM users_workspaces_object WHERE userId = ? [2] []
> [2018-02-07 14:35:41] security.DEBUG: Passing guard token information to the GuardAuthenticationProvider {"firewall_key":"admin","authenticator":"Pimcore\\Bundle\\AdminBundle\\Security\\Guard\\AdminAuthenticator"} []
> [2018-02-07 14:35:41] security.INFO: Guard authentication successful! {"token":"[object] (Symfony\\Component\\Security\\Guard\\Token\\PostAuthenticationGuardToken: PostAuthenticationGuardToken(user=\"admin\", authenticated=true, roles=\"ROLE_PIMCORE_ADMIN\"))","authenticator":"Pimcore\\Bundle\\AdminBundle\\Security\\Guard\\AdminAuthenticator"} []
> [2018-02-07 14:35:41] security.DEBUG: Guard authenticator set no success response: request continues. {"authenticator":"Pimcore\\Bundle\\AdminBundle\\Security\\Guard\\AdminAuthenticator"} []
> [2018-02-07 14:35:41] security.DEBUG: Remember me skipped: it is not configured for the firewall. {"authenticator":"Pimcore\\Bundle\\AdminBundle\\Security\\Guard\\AdminAuthenticator"} []
> [2018-02-07 14:35:41] admin.DEBUG: Perspective for user admin was not requested. Falling back to default. {"user":"admin","perspective":"default"} []
> [2018-02-07 14:35:41] pimcore_admin.session.DEBUG: Opening admin session PHPSESSID {"name":"PHPSESSID"} []
> [2018-02-07 14:35:41] pimcore_admin.session.DEBUG: Admin session PHPSESSID was successfully opened. Open admin sessions: 1 {"name":"PHPSESSID","count":1} []
> [2018-02-07 14:35:41] pimcore_admin.session.DEBUG: Admin session PHPSESSID was written and closed {"name":"PHPSESSID"} []
> [2018-02-07 14:35:41] doctrine.DEBUG: SELECT * FROM `_tmp_update` WHERE revision = ? ["194"] []
> [2018-02-07 14:35:41] pimcore.ERROR: Symfony\Component\Process\Exception\ProcessFailedException: The command "php /usr/local/bin/composer update -n --no-scripts -d /myserverfolder/htdocs" failed.  Exit Code: 1(General error)  Working directory: /myserverfolder/htdocs/web  Output: ================   Error Output: ================                                                                                                 [RuntimeException]                                                                           The HOME or COMPOSER_HOME environment variable must be set for composer to run correctly                                                                                                  in /myserverfolder/htdocs/vendor/symfony/symfony/src/Symfony/Component/Process/Process.php:239 Stack trace: #0 /myserverfolder/htdocs/pimcore/lib/Pimcore/Update.php(433): Symfony\Component\Process\Process->mustRun() #1 /myserverfolder/htdocs/pimcore/lib/Pimcore/Update.php(391): Pimcore\Update::composerUpdate(Array) #2 /myserverfolder/htdocs/pimcore/lib/Pimcore/Update.php(298): Pimcore\Update::installComposerJson('/myserverfolder/h...', '/myserverfolder/h...') #3 /myserverfolder/htdocs/pimcore/lib/Pimcore/Bundle/AdminBundle/Controller/Update/IndexController.php(134): Pimcore\Update::installData('194', 'false') #4 /myserverfolder/htdocs/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php(151): Pimcore\Bundle\AdminBundle\Controller\Update\IndexController->jobProceduralAction(Object(Symfony\Component\HttpFoundation\Request), Object(AppKernel)) #5 /myserverfolder/htdocs/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php(68): Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object(Symfony\Component\HttpFoundation\Request), 1) #6 /myserverfolder/htdocs/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php(202): Symfony\Component\HttpKernel\HttpKernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true) #7 /myserverfolder/htdocs/web/app.php(55): Symfony\Component\HttpKernel\Kernel->handle(Object(Symfony\Component\HttpFoundation\Request)) #8 {main} [] []
> [2018-02-07 14:35:41] doctrine.DEBUG: SELECT id,            (CASE WHEN expire IS NULL OR expire > ? THEN data ELSE NULL END) as data             FROM cache WHERE id IN (?) [1518010541,["system_cache_write_lock"]] []
> [2018-02-07 14:35:41] doctrine.DEBUG: SELECT id,            (CASE WHEN expire IS NULL OR expire > ? THEN data ELSE NULL END) as data             FROM cache WHERE id IN (?) [1518010541,["system_cache_write_lock"]] []



I'd now attempt to run composer install as proposed by dpfaffenbauer.

**Comment by brusch** [comment]
@Cruiser13 

The following log entry seems to point at the actual issue :) 
```
Symfony\Component\Process\Exception\ProcessFailedException: The command "php /usr/local/bin/composer update -n --no-scripts -d /myserverfolder/htdocs" failed. Exit Code: 1(General error) Working directory: /myserverfolder/htdocs/web Output: ================ Error Output: ================ [RuntimeException] The HOME or COMPOSER_HOME environment variable must be set for composer to run correctly in /myserverfolder/htdocs/vendor/symfony/symfony/src/Symfony/Component/Process/Process.php:239 
```

**Comment by brusch** [comment]
@Cruiser13 https://github.com/pimcore/pimcore/commit/9167692e4bcc8f365342089e59b77f2c9532dc80 should fix your issue - can you please try it again after you've applied the patch above? 

Thanks!

**Comment by Cruiser13** [comment]
Hi @brusch they did sadly not fix my issue. Same error  in dev.log:
pimcore.ERROR: Symfony\Component\Process\Exception\ProcessFailedException: The command "php /usr/local/bin/composer update -n --no-scripts -d /myserverfolder/htdocs" failed.  Exit Code: 1(General error)  Working directory:...

I did run the command which failed for symfony through SSH which resulted in this:
web1@vs10972:~$ php /usr/local/bin/composer update -n --no-scripts -d /myserverfolder/htdocs
Cannot create cache directory /myserverfolder/.composer/cache/repo/https---packagist.org/, or directory is not writable. Proceeding without cache
Cannot create cache directory /myserverfolder/.composer/cache/files/, or directory is not writable. Proceeding without cache
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Installing presta/sitemap-bundle (v1.5.2)
    Downloading: 100%

Writing lock file
Generating optimized autoload files

I was able to get into pimcore afterwards and restart the 5.1.3 update. The 5.1.3 update went smooth but I had to run composer update manually afterwards through command line.

Composer is defined in the parameters.yml by the way, not sure if that matters:
pimcore_executable_composer: php /usr/local/bin/composer.phar

**Comment by brusch** [comment]
> Composer is defined in the parameters.yml by the way, not sure if that matters:
pimcore_executable_composer: php /usr/local/bin/composer.phar

Yes, that matters, thanks for the hint. 
Commit 76b80ea67360c59e073e6a58d5ec0302ca6270eb should fix your problem.

**Comment by Cruiser13** [comment]
Hi @brusch thank you but it seems to stay complicated. If I apply all three patches to a clean 5.1.0 pimcore and install with empty profile, I'll end up with this error message while trying to update to 5.1.3:


> Server threw exception - could not perform action. Please reload the admin interface and try again.
> 
> Timestamp: Mon Feb 12 2018 15:18:47 GMT+0100
> Status: 500 | Internal Server Error
> URL: /admin/update/index/job-procedural
> Message: You have requested a non-existent service "Pimcore\Composer\Config\ConfigMerger". Did you mean this: "Pimcore\Controller\Config\ConfigNormalizer"


If I do not change the Update.php, but apply the two other patches, I'll end up with the missing PrestaSitemapBundle again.

**Comment by SmashingQuasar** [comment]
So I did run composer update, it worked and then I installed 5.1.3.

Everything went smooth during installation. But right now everything is totally broken. Objects are displayed in the back-office, but there respective tables are all empty.

E.g. : I have a class which has the id 2 named Offer. I had around 1K5 Offers registered before the update. Now they still all exist in the back-office, but they are all empty. When I'm checking tables object_query_2, object_store_2 and object_2, they all return an empty set. However when checking the objects table, all the objects are registered.

I can't even delete them from the back-office, because it seems you changed the way object bricks work, and switched the table names to lowercase, while they were to upper camel case (for object bricks only) before, now the system is trying to look in the table using the upper camel case name, which obviously doesn't work since MariaDB is case-sensitive.

I'm using MariaDB 10.1.
